src/test/java/io/axoniq/axonserver/connector/query/QueryChannelIntegrationTest.java

```diff
@@ -92,6 +95,35 @@ void tearDown() {
         connectionFactory2.shutdown();
     }
 
+    @Test
+    void testPermitsAreRenewedForSubscriptionQuery() throws Exception {
+        QueryChannel queryChannel = connection1.queryChannel();
+        queryChannel.registerQueryHandler(new SubscriptionQueryHandler(), new QueryDefinition("testQuery", "testResult"))
+                    .awaitAck(1, TimeUnit.SECONDS);
+        queryChannel.registerQueryHandler(new SubscriptionQueryHandler(), new QueryDefinition("testQuery", "testResult"))
+                    .awaitAck(1, TimeUnit.SECONDS);
+
+        QueryChannel queryChannel2 = connection2.queryChannel();
+        queryChannel2.registerQueryHandler(new SubscriptionQueryHandler(), new QueryDefinition("testQuery", "testResult"))
+                     .awaitAck(1, TimeUnit.SECONDS);
+
+        List<CompletableFuture<?>> results = new ArrayList<>();
+
+        // 5000 will produce well more messages than available permits.
```

Comment:
Although the comment states 5000, the for-loop increments untill 1000.

Reply:
There is another PR open, where the number of permits is explicitly configured to 100. We didn't want to add that possibility in this PR as well, to avoid pollution.

Reply:
Awesome plan.

Reply:
Awesome plan.

Reply:
Awesome plan.